<ui version="4.0" >
 <class>ReportWidget</class>
 <widget class="QWidget" name="ReportWidget" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>357</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Coverage Report</string>
  </property>
  <property name="whatsThis" >
   <string>GCOV code coverage statistics.

Set the directory to generate the report for and start the scan. 

Once finished, you can select the directories in the coverage report to see their statistics. A more detailed report can be seen for each directory, showing all the covered files in it. Each file can also be shown in the editor, annotated with line by line coverage information.</string>
  </property>
  <layout class="QVBoxLayout" name="reportWidgetLayout" >
   <item>
    <widget class="QFrame" name="buildPathFrame" >
     <property name="frameShape" >
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="buildPathFrameLayout" >
      <item>
       <widget class="QLabel" name="buildPathTextLabel" >
        <property name="whatsThis" >
         <string>The directory to scan for .gcda files.</string>
        </property>
        <property name="text" >
         <string>Build Path: </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="startButton" >
        <property name="toolTip" >
         <string>Start coverage analysis</string>
        </property>
        <property name="whatsThis" >
         <string>Start coverage analysis in the specified path.

The directory and, recursively, its subdirectories are scanned for .gcda files.</string>
        </property>
        <property name="text" >
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="statisticsFrame" >
     <property name="frameShape" >
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="statisticsFrameLayout" >
      <item>
       <spacer name="horizontalSpacer_3" >
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>71</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout" >
        <item>
         <widget class="KLineEdit" name="filterBox" >
          <property name="toolTip" >
           <string>Filter directories by name</string>
          </property>
          <property name="whatsThis" >
           <string>When the report is in directory mode, only those containing the specified text in their path are shown.

In file mode, the filter has no effect.</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" >
          <item>
           <spacer name="horizontalSpacer" >
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>30</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout" >
            <item row="0" column="0" >
             <widget class="QLabel" name="coverageRatioTextLabel" >
              <property name="whatsThis" >
               <string>Percentage of covered lines in the selected directories.</string>
              </property>
              <property name="text" >
               <string>Line Coverage:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QLabel" name="coverageRatio" >
              <property name="whatsThis" >
               <string>Percentage of covered lines in the selected directories.</string>
              </property>
              <property name="text" >
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" >
             <widget class="QLabel" name="nrofCoveredLinesTextLabel" >
              <property name="whatsThis" >
               <string>Number of covered lines in the selected directories.

Covered lines are those that were executed at least once.</string>
              </property>
              <property name="text" >
               <string>Covered Lines:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1" >
             <widget class="QLabel" name="nrofCoveredLines" >
              <property name="whatsThis" >
               <string>Number of covered lines in the selected directories.

Covered lines are those that were executed at least once.</string>
              </property>
              <property name="text" >
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" >
             <widget class="QLabel" name="slocTextLabel" >
              <property name="whatsThis" >
               <string>Total number of lines of code in the selected directories.

Note that it only includes executable lines. That is, things like comments, macros, forward declarations... aren't taken into account.</string>
              </property>
              <property name="text" >
               <string>Total SLOC:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1" >
             <widget class="QLabel" name="sloc" >
              <property name="whatsThis" >
               <string>Total number of lines of code in the selected directories.

Note that it only includes executable lines. That is, things like comments, macros, forward declarations... aren't taken into account.</string>
              </property>
              <property name="text" >
               <string>-</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_2" >
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>30</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_4" >
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>71</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Veritas::DrillDownView" name="table" >
     <property name="whatsThis" >
      <string>Coverage report.

In directory mode, the report summarizes the coverage with a list of all the directories that contain at least one covered file. It could include directories out of the specified path, depending on the files included in the source code.

In file mode, all the files of the selected directory are shown. If several directories are selected, only the files of the last selected directory are shown. Double clicking on any of the items shows the associated file in the editor, annotated with the coverage information.

You can change between directory and file mode using left and right keys.</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KLineEdit</class>
   <extends></extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>Veritas::DrillDownView</class>
   <extends>QTableView</extends>
   <header>drilldownview.h</header>
   <slots>
    <signal>completedSlideLeft()</signal>
    <signal>completedSlideRight()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>startButton</tabstop>
  <tabstop>filterBox</tabstop>
  <tabstop>table</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
